src/lib/session-management.js

```diff
@@ -0,0 +1,200 @@
+    SessionManagement.instance = this;
+  }
+
+  static getInstance() {
+    if (!SessionManagement.instance) {
+      SessionManagement.instance = new SessionManagement();
+    }
+    return SessionManagement.instance;
+  }
+
+  init(config = {}) {
+    if (typeof config !== 'object') {
+      throw new Error('[LIBRARY] Invalid configuration object');
+    }
+    this.config = Object.freeze({ ...defaultConfig, ...config });
+    console.debug('[LIBRARY] Initialising session management with config:', this.config);
+  }
+
+  setSessionExpiryTime(sessionExpiryTime, refreshExpiryTime) {
+    console.debug('[LIBRARY] Setting session expiry time');
+    this.initialiseSessionExpiryTimers(sessionExpiryTime, refreshExpiryTime);
+  }
+
+  async initialiseSessionExpiryTimers(sessionExpiryTime, refreshExpiryTime) {
+    console.debug('[LIBRARY] init config: ', this.config);
+    if (!this.config || Object.keys(this.config).length === 0) {
+      console.debug('[LIBRARY] No config found, initialising with default config');
+      this.init(this.config);
+    }
+
+    try {
+      console.debug('[LIBRARY] Checking initial session state');
+      const { checkedSessionExpiryTime, checkedRefreshExpiryTime } = await checkSessionStatus();
+
+      const finalSessionExpiryTime = checkedSessionExpiryTime || validateExpiryTime(sessionExpiryTime);
+      const finalRefreshExpiryTime = checkedRefreshExpiryTime || validateExpiryTime(refreshExpiryTime);
+
+      console.debug('[LIBRARY] sessionExpiryTime: ', finalSessionExpiryTime);
+      console.debug('[LIBRARY] refreshExpiryTime: ', finalRefreshExpiryTime);
+
+      if (finalSessionExpiryTime) {
+        console.debug(`[LIBRARY] Session expiry time: ${finalSessionExpiryTime}`);
+        this.startSessionTimer(finalSessionExpiryTime);
+      }
+      if (finalRefreshExpiryTime) {
+        console.debug(`[LIBRARY] Refresh expiry time: ${finalRefreshExpiryTime}`);
+        this.startRefreshTimer(finalRefreshExpiryTime);
+      }
+      if (!finalSessionExpiryTime && !finalRefreshExpiryTime) {
+        console.error('[LIBRARY] Failed to initialise session expiry timers: No expiry times provided');
+        this.handleSessionValidity(false);
+      } else {
+        this.handleSessionValidity(true, finalSessionExpiryTime, finalRefreshExpiryTime);
+      }
+    } catch (error) {
+      console.error('[LIBRARY] Failed to initialise session expiry timers:', error);
+      this.handleSessionValidity(false);
+    }
+  }
+
+  handleSessionValidity(isValid, sessionExpiryTime, refreshExpiryTime) {
+    const { onSessionValid, onSessionInvalid } = this.config;
+
+    if (isValid) {
+      if (onSessionValid) {
+        onSessionValid(sessionExpiryTime, refreshExpiryTime);
+      } else {
+        console.debug('[LIBRARY] No onSessionValid callback provided.');
+      }
+    } else if (onSessionInvalid) {
+      onSessionInvalid();
+    } else {
+      console.debug('[LIBRARY] No onSessionInvalid callback provided.');
+    }
+  }
```

Comment:
```suggestion
    if (isValid) {
      if (onSessionValid) {
        onSessionValid(sessionExpiryTime, refreshExpiryTime);
      } else {
        console.debug('[LIBRARY] No onSessionValid callback provided.');
      }
    } else {
      if (onSessionInvalid) {
        onSessionInvalid();
      } else {
        console.debug('[LIBRARY] No onSessionInvalid callback provided.');
      }
    }
  }
```

I get that this logic works, however I think it would be clearer like this.

Reply:
or could do the sub ifs as ternaries.